state/executor.go

```diff
@@ -123,6 +129,79 @@ type BlockResult struct {
 	TotalGas uint64
 }
 
+// getStakingEventType is a helper method for getting the type of system event
+func getStakingEventType(toAddress types.Address) staking.StakingEventType {
+	if toAddress == types.StringToAddress(system.StakingAddress) {
+		return staking.StakingEvent
+	}
+
+	return staking.UnstakingEvent
```

Comment:
I think that this method (by itself) shouldn't assume that anything that's passed into it is either staking or unstaking address. It should check if address is unstaking one, and if not return either error or something like "unknown event"

Reply:
Got it. Added a new event type for unknown events, and now handle them accordingly 👍